src/12-bluetooth-setup/at-commands.md

```diff
@@ -1,43 +1,60 @@
 ## AT commands
 
-> **NOTE** incomplete
+The Bluetooth module and the F3 need to be configured to communicate at the same baud rate. The tutorial code initializes the UART1 serial device to a baud rate of 115200. The HC-05 Bluetooth module is configured at a baud rate of 9600 by default.
+
+The Bluetooth module supports an AT mode that allows you to examine and change its configuration and settings. To utilize the AT mode, connect the Bluetooth module to the F3 and FTDI as shown in the following diagram.
 
 <p align="center">
 <img height=640 title="Bluetooth <-> Serial connection" src="assets/bluetooth-serial.png">
 </p>
 
-Entering AT mode:
-
-- Power off the Bluetooth module
-- Press and hold the button on the Bluetooth module
-- Power on the Bluetooth module
-- Now, release the button
+Recommended steps to enter AT mode:
 
-> **TODO** blinking pattern
+- Disconnect the F3 and FTDI from your laptop.
+- Connect F3's GND pin to the Bluetooth's GND pin using a Female/Female (F/F) wire
+  (preferably, a black one).
+- Connect F3's 5V pin to the Bluetooth's VCC pin using a F/F wire (preferably, a
+  red one).
+- Connect the FTDI RXI pin to the Bluetooth's TXD pin using a Female/Male (F/M) wire.
+- Connect the FTDI TXO pin to the Bluetooth's RXD pin using a Female/Male (F/M) wire.
+- Now connect the FTDI to your laptop via USB cable.
+- Next connect the F3 to your laptop via USB cable while simultaneously pressing and holding the button on the Bluetooth module (kinda tricky).
+- Now, release the button and the Bluetooth module will enter AT mode. You can confirm this by observing that the red LED on the Bluetooth module is blinking in a slow pattern (approx 1-2 seconds on/off).
 
-The AT mode always operates at a baud rate of 38,400.
+The AT mode always operates at a baud rate of 38400, so configure minicom/putty for that baud rate and connect to the FTDI device.
```

Comment:
```suggestion
The AT mode always operates at a baud rate of 38400, so configure your terminal program for that baud rate and connect to the FTDI device.
```

I think we should not refer to specific applications here, otherwise we'll have to revise all chapters all the time whenever we suggest the use of a new application.

Reply:
Can't apply this from GUI, so pushed with separate commit. Thanks!